### DISCUSSION TITLE
ERROR (Mrsk::Utils::HealthcheckPoller::HealthcheckError): Exception while executing on host x.x.x.x: container not ready (unhealthy)

### DISCUSSION THREAD

**Discussion by RayNawara** [basecamp/kamal#365] [General]
Hi!

I read the similar discussion but my issue is different. I believe my app is coming up correctly but it just doesn't tell MRSK.
Rails 7.0.4. App builds fine on my local.

Here is my error:

` INFO container not ready (unhealthy), retrying in 7s (attempt 7/7)...
  INFO [0cfa3f3b] Running docker container ls --all --filter name=^healthcheck-mba-ai-b899707602d62df5367770f4a952b2163c882703$ --quiet | xargs docker inspect --format '{{if .State.Health}}{{.State.Health.Status}}{{else}}{{.State.Status}}{{end}}' on 5.78.87.35
  INFO [0cfa3f3b] Finished in 0.200 seconds with exit status 0 (successful).
  INFO [082e98c2] Running docker container ls --all --filter name=^healthcheck-mba-ai-b899707602d62df5367770f4a952b2163c882703$ --quiet | xargs docker logs --tail 50 2>&1 on 5.78.87.35
  INFO [082e98c2] Finished in 0.325 seconds with exit status 0 (successful).
 ERROR Puma starting in single mode...
* Puma version: 6.1.0 (ruby 3.2.2-p53) ("The Way Up")
*  Min threads: 5
*  Max threads: 5
*  Environment: development
*          PID: 17
* Listening on http://0.0.0.0:3000
Use Ctrl-C to stop
 ERROR {
  "Status": "unhealthy",
  "FailingStreak": 26,
  "Log": [
    {
      "Start": "2023-06-28T16:51:35.374337219Z",
      "End": "2023-06-28T16:51:35.522233081Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0 75160    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 404\n"
    },
    {
      "Start": "2023-06-28T16:51:36.525212272Z",
      "End": "2023-06-28T16:51:36.675101381Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0 75160    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 404\n"
    },
    {
      "Start": "2023-06-28T16:51:37.678332622Z",
      "End": "2023-06-28T16:51:37.834537293Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0 75160    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 404\n"
    },
    {
      "Start": "2023-06-28T16:51:38.837293521Z",
      "End": "2023-06-28T16:51:39.003126106Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0 75160    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 404\n"
    },
    {
      "Start": "2023-06-28T16:51:40.00643425Z",
      "End": "2023-06-28T16:51:40.173779523Z",
      "ExitCode": 1,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0 75160    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (22) The requested URL returned error: 404\n"
    }
  ]
}
  INFO [a3b5f323] Running docker container ls --all --filter name=^healthcheck-mba-ai-b899707602d62df5367770f4a952b2163c882703$ --quiet | xargs docker stop on 5.78.87.35
  INFO [a3b5f323] Finished in 10.526 seconds with exit status 0 (successful).
  INFO [94a2a7fc] Running docker container ls --all --filter name=^healthcheck-mba-ai-b899707602d62df5367770f4a952b2163c882703$ --quiet | xargs docker container rm on 5.78.87.35
  INFO [94a2a7fc] Finished in 0.358 seconds with exit status 0 (successful).
  Finished all in 171.2 seconds
Releasing the deploy lock...
  Finished all in 173.2 seconds
  ERROR (Mrsk::Utils::HealthcheckPoller::HealthcheckError): Exception while executing on host xx.xx.xx.x: container not ready (unhealthy)`

I tried putting 
`get "up" => "rails/health#show", as: :rails_health_check`
in my routes.rb but no help.

I know I missed something stupid but I can't see it.

(Btw, I think the 404 is from traefik.)

Thanks!

**Comment by Jberczel** [comment]
I see you added route to Rails 6 app.  Did you also create the controller and action as well?

Alternatively, do you have any other static page like an `/about` page?  If so, you could configure your docker healthcheck to use that in your `config/deploy.yml` file.

**Reply by RayNawara** [reply]
Thanks. As I said above, it's a rails 7.0.4 app. But I'll try your idea.

Thanks again!

**Comment by RayNawara** [comment]
@Jberczel I just changed my healtcheck to / and everything works now. Thanks!
